script/DeployHypMinter.s.sol

```diff
@@ -0,0 +1,170 @@
+            _distributionDelay: 7 days,
+            _operatorRewardsManager: 0x562Dfaac27A84be6C96273F5c9594DA1681C0DA7 // Multisig A
+        });
+    }
+
+    function run(
+        AccessManager _accessManager,
+        uint256 _firstRewardTimestamp,
+        uint256 _mintAllowedTimestamp,
+        uint256 _distributionAllowedTimestamp,
+        uint256 _distributionDelay,
+        address _operatorRewardsManager
+    ) public {
+        require(_distributionDelay <= DISTRIBUTION_DELAY_MAXIMUM, "Distribution delay too large");
+        require(_mintAllowedTimestamp > block.timestamp, "Mint allowed timestamp must be in future");
+        require(_distributionAllowedTimestamp > block.timestamp, "Distribution allowed timestamp must be in future");
+
+        DeployConfig memory config = DeployConfig({
+            firstRewardTimestamp: _firstRewardTimestamp,
+            mintAllowedTimestamp: _mintAllowedTimestamp,
+            distributionAllowedTimestamp: _distributionAllowedTimestamp,
+            distributionDelay: _distributionDelay,
+            accessManager: _accessManager,
+            operatorRewardsManager: _operatorRewardsManager
+        });
+
+        _deploy(config);
+    }
+
+    function _deploy(
+        DeployConfig memory config
+    ) internal {
+        // Start broadcast
+        address deployer = _getDeployer();
+        vm.startBroadcast(deployer);
+
+        // Deploy implementation
+        console2.log("Deploying HypMinter implementation...");
+        HypMinter implementation = new HypMinter(DISTRIBUTION_DELAY_MAXIMUM);
+        console2.log("Implementation deployed at:", address(implementation));
+
+        // Prepare initialization data
+        bytes memory initData = abi.encodeCall(
+            HypMinter.initialize,
+            (
+                AccessManager(config.accessManager),
+                config.firstRewardTimestamp,
+                config.mintAllowedTimestamp,
+                config.distributionAllowedTimestamp,
+                config.distributionDelay,
+                config.operatorRewardsManager
+            )
+        );
+
+        // Deploy proxy with deployer as proxy admin
+        console2.log("Deploying TransparentUpgradeableProxy...");
+        TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(
+            address(implementation),
+            deployer, // Use deployer as proxy admin
+            initData
+        );
+        console2.log("Proxy deployed at:", address(proxy));
+
+        // Get the proxied contract
+        HypMinter hypMinter = HypMinter(address(proxy));
+
+        vm.stopBroadcast();
+
+        // Log deployment summary
+        logDeploymentSummary(hypMinter, implementation, config);
+
+        // Verify deployment
+        verifyDeployment(hypMinter, config);
+    }
+
+    function logDeploymentSummary(HypMinter hypMinter, HypMinter implementation, DeployConfig memory config) internal {
```

Comment:
this is more of a style comment but I personally dont think this is that useful
foundry already has its own callstack logs and artifacts format

Reply:
Removed